enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,244 @@
+last-updated: 2021-12-05
+status: implementable
+---
+
+# containerd - new container runtime  
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [OpenShift-docs](https://github.com/OpenShift/OpenShift-docs/)
+
+## Summary
+
+This enhancement intends to allow customers to bring up Windows nodes with containerd
+as the default runtime from OpenShift 4.11 onwards. When customers upgrade clusters to
+OpenShift 4.11, container runtime will be migrated from Docker to containerd
+
+## Motivation
+
+In Kubernetes, the container runtime interface has been used to talk to a container runtime. From
+Kubernetes 1.24 onwards dockershim will be removed from kubelet code. At present
+WMCO uses Docker as the default runtime. The goal is to make containerd the default
+runtime and move away from Docker before dockershim has been decoupled from kubelet.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Make containerd the default container runtime on Windows nodes.
+* When WMCO is upgraded from 5.x.z to 6.x.z, the runtime on all existing Windows nodes will be migrated to
+  containerd.  
+
+### Non-Goals
+
+* Uninstalling Docker runtime is not part of this enhancement.
+* Allowing the user to choose a container runtime is not supported.
+* Downgrade is not supported.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+## Proposal
+
+To make containerd the default runtime, containerd should be started as a Windows
+service before starting kubelet as Windows service. While starting kubelet, the parameters `container-runtime` and
+`container-runtime-endpoint` should be configured. The remaining WMCO workflow of bringing Windows nodes into the
+Kubernetes cluster has no impact other than the proposed changes. MachineSet and BYOH (Bring Your Own Host) upgrade
+details are discussed in the upgrade section. There is no difference between MachineSet and BYOH on how we enable
+containerd as the default runtime.
+
+Golden Image:
+
+vSphere and BYOH use golden image to create Windows VM. Customers can create a new Windows golden image excluding Docker
+runtime or continue to use the same Windows image with Docker. Once containerd becomes the default runtime, WMCO will
+support Windows golden image with or without Docker.
+
+Containerd Migration plan:
+
+Containerd will be the default runtime in OpenShift 4.11 for Windows. For WMCO development and testing, we plan
+to introduce a WMCO CLI option which allows us to choose the runtime. This CLI option will be used in the
+release of the WMCO community operator for OpenShift 4.10 to allow customers to try out Windows
+nodes with containerd. Customers can use 4.10 community WMCO to create new clusters or upgrade
+existing clusters.
```

Comment:
>Customers can use 4.10 community WMCO to create new clusters or upgrade existing clusters.

We don't support upgrades across product and community operators. I suggest removing this line.